### ISSUE TITLE
[Site Request] kmanga

### ISSUE THREAD

**Issue by TristanWasTaken** [manga-download/hakuneko#5866]
**Name of the website**
K MANGA

**Website urls (examples below)**
- Site: https://kmanga.kodansha.com
- Manga List: https://kmanga.kodansha.com/ranking (/series doesn't include all manga)
- Manga example: https://kmanga.kodansha.com/title/10136/episode/312468
- Chapter online viewer example: https://kmanga.kodansha.com/title/10136/episode/312468
- Known domains:
  - api https://api.kmanga.kodansha.com/
    - manga list example: https://api.kmanga.kodansha.com/ranking/all?version=6.0.0&platform=3&offset=0&limit=101
    - manga example: https://api.kmanga.kodansha.com/title/list?version=6.0.0&platform=3&title_id_list=10136
    - chapter list example (POST request): https://api.kmanga.kodansha.com/episode/list `{"version":"6.0.0","platform":"3","episode_id_list":"id1,id2,id3"}`
    - chapter example (includes cdn image urls): https://api.kmanga.kodansha.com/web/episode/viewer?version=6.0.0&platform=3&episode_id=312468
  - cdn https://cdn.kmanga.kodansha.com/
    - image url example https://cdn.kmanga.kodansha.com/static/web_titles/10136/episodes/312468/56ef39e0b2213064dfb82f27807cd12201c9c44b7bc16b1f384bd8ec7a62cd54.jpg?Expires=1685621974&Signature=hvKjALOUn~097spmSXUHp0rs10AsX0zLgqqVVWD6WGQ2hv1PeOIJ5v7GO7D81LkpSEqR6Rw1aNPWKKRXDDJw5tJDv39VYMXJnfg4Ej~Kl2dbQOeN1vpIsgV30Qw3A8s04XjxQBI1MRCL~wEBadnvoQHjd9n9rLShpNZQ7Q67rN~Oz32wCL-YvPg81NhmzJabFCPcLoxm0zCLbZvMUAi6hcg3O6HloyOXB7n6OFzxs8HTwMAomaJjFd4CwP9DObg-VkKqVDttSeO0gbGokqMfiyjgrdMV4ayDrrTRAlFB2TqFNCMR1UOGQYO1nsMEpMVHW9HdXHniPOnUaTH6DYHmuA__&Key-Pair-Id=K25J5B1H7MBQ0O

**Languages**
English

**Website relationship*
site design and layout with general infrastructure is *very* similar to https://pocket.shonenmagazine.com/

**Additional details**
- Kmanga seems to use image scrambling for the chapter images
- They block a lot of locations (the us should be unblocked). They seem to block some well known vpn ip addresses. If you got the block page, the browser site data has to be reset before switching the vpn ip address since they seem to track it.
- Please note that it's against kmanga's [tos](https://kmanga.kodansha.com/article/terms-of-service) to "access, search, or meta-search the Services"
> l). Attempt to access, search, or meta-search the Services with any engine, software, tool, agent, device, or mechanism other than software and/or search agents provided by Kodansha or other generally available third-party web browsers (such as Microsoft Edge, Google Chrome, Mozilla Firefox, Safari or Opera), including, without limitation, any software that sends queries to the Services to determine how a website or web page ranks;

**Comment by TristanWasTaken** [comment]
it seems like it's possible to open the general site layout outside of the us by opening the [tos](https://kmanga.kodansha.com/article/terms-of-service) page and then clicking on the top left logo

**Comment by MikeZeDev** [comment]
> site design and layout with general infrastructure is very similar to https://pocket.shonenmagazine.com/

` export default class ShonenMagazinePocket extends CoreView {` ,

Maybe try do it  by extending CoreView then ? :)

**Comment by TristanWasTaken** [comment]
> ` export default class ShonenMagazinePocket extends CoreView {` ,
> 
> Maybe try do it by extending CoreView then ? :)

@MikeZeDev nope, coreview doesn't work. The site is sadly only similar and not the same. Besides mostly the same layout, it has an api, an actual ranking page (and not just a section on the main page), and doesn't use ssr like pocket. kmanga is from what I see a spa, which is probably why they have an api

**Comment by devmatcha** [comment]
couldn't get coreview to work either

**Comment by valdearg** [comment]
Not sure if it's any help, but there's a scramble_key in the viewer response:

https://api.kmanga.kodansha.com/web/episode/viewer?version=6.0.0&platform=3&episode_id=343330

And the code is in the JSON in: https://kmanga.kodansha.com/assets/chunk-6010e0fd.js

**Comment by aftercares0** [comment]
I've (well, Opus 4.6) figured out how to unscramble the image. Here's the analysis.

[scrambling_algorithm_analysis.md](https://github.com/user-attachments/files/27852327/scrambling_algorithm_analysis.md)

**Comment by MikeZeDev** [comment]
Similar to Ciaoplus and ShonenMagazine.
